evalml/automl/automl_algorithm/iterative_algorithm.py

```diff
@@ -242,6 +241,20 @@ def _transform_parameters(self, pipeline, proposed_parameters):
                         )
                     else:
                         component_parameters[param_name] = value
+            if name in self._pipeline_params and self._batch_number == 0:
+                for param_name, value in self._pipeline_params[name].items():
```

Comment:
Tagging on to what @dsherry mentioned below, we may want to do a check for the "reverse" - specifying a custom hyperparameter without using `skopt.Space`. I think this may be helpful because not using `skopt.Space` for custom hyperparameters will not work as intended. I think it's an skopt issue since the tuner is properly created! 

```python
from evalml import AutoMLSearch
from evalml.model_family import ModelFamily
from evalml.demos import load_fraud


X, y = load_fraud(1000)

automl = AutoMLSearch(X, y, problem_type="binary", allowed_model_families=[ModelFamily.RANDOM_FOREST],
                      custom_hyperparameters={"Random Forest Classifier": {"n_estimators": 5}},
                      max_batches=3)
automl.search()
n_estimators_used = automl.full_rankings.parameters.iloc[:11].map(lambda d: d["Random Forest Classifier"]['n_estimators'])
assert set(n_estimators_used) != {5}
```

![image](https://user-images.githubusercontent.com/41651716/120867734-842ce880-c560-11eb-9298-e994ea45f385.png)

Reply:
Great point!